src/domain/data/index.ts

```diff
@@ -0,0 +1,44 @@
+export class DataProvider {
+  private readonly encoder = new TextEncoder();
+  constructor(
+    private readonly textLen: number,
```

Comment:
What is the purpose of private here?

Reply:
It defines the total text length for the provider to generate on each `getData` call (line 9). A DataProvider instance is meant to be reusable to generate many instances of the same problem size (both text length and pattern length).

Reply:
Alright, I wasn't precise :D, I had concerns about "private", not the textLen itself. So now I know that's the inline definition of a class field, right? If so it can be as it is.

Reply:
Yes. This is a syntactic shorthand to setting the provided parameter as a private class member.
Resolving conversation.